[8.x] Ability to return the default value of a request whenHas and whenFilled methods

olekjs opened the pull request:
Many times I needed to return the default value using one of the `whenHas` or `whenFilled` methods, but I could not which was annoying because such a great method was practically useless in some cases.

The solution taken from the collection, we have there for example the `whenEmpty` method, which takes a paramter that returns the default value.

Let's see how this is used in collections:

```php
$collection->whenEmpty(function ($collection) {
    return $collection->push('Adam');
}, function ($collection) {
    return $collection->push('Taylor');
});
```

In the same way now it would be possible to obtain in the request object:

```php
$request->whenHas('name', function ($name) {
    return $name;
}, function () {
    return 'some default value';
});
```

If we do not pass the default value, we will get an instance of the request class, which is the same as before the changes I made.

If I was the only person who needed this solution, then PR is to close.

I also added tests and I hope that I considered each case, I tried to test the introduced code so that it does not force changes on existing solutions.

Commit:
Add the ability to return the default value of a request whenHas and whenFilled methods

taylorotwell commented on src/Illuminate/Http/Concerns/InteractsWithInput.php:
```diff
@@ -111,13 +111,24 @@ public function hasAny($keys)
+        $value = data_get($this->all(), $key);
+
         if ($this->has($key)) {
-            return $callback(data_get($this->all(), $key)) ?: $this;
```
You have changed the entire behavior of the function. Just add the default support. Don't change the behavior.

taylorotwell commented:
You have changed the entire behavior of the function regarding returning `$this`, etc. Please correct that.

olekjs committed:
Merge branch 'laravel:8.x' into 8.x

Commit:
Change functionality and test fixes

olekjs commented:
@taylorotwell You're right. I already changed the code. Now there is only support for the default value. In addition, I removed the value parameter from default, because it will always be null, so it was unnecessarily there.

Tests have also been slimmed down, if anything else will be to correct let me know!
